services/wallet/activity/activity.go

```diff
@@ -652,6 +655,9 @@ func getActivityEntries(ctx context.Context, deps FilterDependencies, addresses
 				involvedToken = &Token{TokenType: Native, ChainID: common.ChainID(chainID.Int64)}
 			}
 			if activityType == SendAT {
+				if toAddress == ZeroAddress && involvedToken.TokenType == Native && *contractAddress != ZeroAddress {
```

Comment:
some DB entries might have a `NULL` `contractAddress` I believe, better to check that `contractAddress` is not `nil` as well. Also, we're doing the same check in two different places, probably a good idea to move it to a separate function